Doc/library/re.rst

```diff
@@ -1565,21 +1565,27 @@ search() vs. match()
 
 .. sectionauthor:: Fred L. Drake, Jr. <fdrake@acm.org>
 
-Python offers two different primitive operations based on regular expressions:
-:func:`re.match` checks for a match only at the beginning of the string, while
-:func:`re.search` checks for a match anywhere in the string (this is what Perl
-does by default).
+Python offers different primitive operations based on regular expressions:
+
++ :func:`re.search` checks for a match anywhere in the string,
+  (this is what Perl does by default)
++ :func:`re.fullmatch` checks for entire string to be a match
++ :func:`re.match` checks for a match only at the beginning of the string
+
 
 For example::
 
-   >>> re.match("c", "abcdef")    # No match
-   >>> re.search("c", "abcdef")   # Match
-   <re.Match object; span=(2, 3), match='c'>
+   >>> re.match("c", "cdef")    # match
+   <re.Match object; span=(0, 1), match='c'>
+   >>> re.search("c", "cdef")   # Match
+   <re.Match object; span=(0, 1), match='c'>
+   >>> re.fullmatch("c", "cdef")  # No Match
```

Comment:
Sorry I wasn't clear. I think you need two examples for fullmatch:
```python
re.fullmatch("p.*n", "python")  # Match
re.fullmatch("r.*n", "python") # No match
```

Reply:
yes sir

Reply:
done sir